shared_model/validators/transactions_collection/transactions_collection_validator.cpp

```diff
@@ -19,22 +19,28 @@
 namespace shared_model {
   namespace validation {
 
-    template <typename TransactionValidator>
-    TransactionsCollectionValidator<TransactionValidator>::
+    template <typename TransactionValidator, bool CollectionCanBeEmpty>
+    TransactionsCollectionValidator<TransactionValidator,
+                                    CollectionCanBeEmpty>::
         TransactionsCollectionValidator(
             const TransactionValidator &transactions_validator)
         : transaction_validator_(transactions_validator) {}
 
-    template <typename TransactionValidator>
+    template <typename TransactionValidator, bool CollectionCanBeEmpty>
     template <typename Validator>
-    Answer TransactionsCollectionValidator<TransactionValidator>::validateImpl(
-        const interface::types::TransactionsForwardCollectionType &transactions,
-        Validator &&validator) const {
+    Answer TransactionsCollectionValidator<TransactionValidator,
+                                           CollectionCanBeEmpty>::
+        validateImpl(const interface::types::TransactionsForwardCollectionType
+                         &transactions,
+                     Validator &&validator) const {
       Answer res;
       ReasonsGroupType reason;
       reason.first = "Transaction list";
 
       if (boost::empty(transactions)) {
```

Comment:
`if (boost::empty(transactions) and not CollectionCanBeEmpty)` looks simpler than a nested `if`